client-linuxapp/src/main.rs

```diff
@@ -446,36 +439,55 @@ async fn main() -> Result<()> {
     }
 
     // Get rv entries
-    let rv_info = get_rv_info(dc.as_ref());
-
-    let client_list = get_client_list(rv_info.unwrap())
-        .await
-        .context("Error getting usable rendezvous client")?;
-
-    // Get owner info
-    let to1d = get_to1d(dc.as_ref(), client_list).await?;
-
-    let to1d_payload: UnverifiedValue<TO1DataPayload> = to1d
-        .get_payload_unverified()
-        .context("Error getting the TO2 payload")?;
-    let to2_addresses = to1d_payload.get_unverified_value().to2_addresses();
-    let to2_addresses = get_to2_urls(to2_addresses);
-    log::info!("Got TO2 addresses: {:?}", to2_addresses);
+    let rv_info = get_rv_info(dc.as_ref())?;
+    let mut rv_entry_it = rv_info.iter().cycle();
 
-    if to2_addresses.is_empty() {
-        bail!("No valid TO2 addresses received");
-    }
-
-    for to2_address in to2_addresses {
-        match perform_to2(devcred_location.borrow(), dc.as_ref(), &to2_address)
+    loop {
+        let rv_entry = rv_entry_it.next().unwrap();
```

Comment:
@puiterwijk I know you're not a big fan of `unwrap`, but tried with `?` here, didn't work, and couldn't find other way to get this done.